lib/bitex_bot/robot.rb

```diff
@@ -9,110 +9,152 @@
 end
 
 module BitexBot
+  ##
+  # Documentation here!
+  #
+  # rubocop:disable Metrics/ClassLength
   class Robot
     cattr_accessor :graceful_shutdown
     cattr_accessor :cooldown_until
-    cattr_accessor :taker do
-      case Settings.taker
-      when 'itbit'
-        ItbitApiWrapper
-      when 'bitstamp'
-        BitstampApiWrapper
-      when 'bitfinex'
-        BitfinexApiWrapper
-      when 'kraken'
-        KrakenApiWrapper
-      end
-    end
+    cattr_accessor(:taker) { "#{Settings.taker.capitalize}ApiWrapper".constantize }
+    cattr_accessor(:current_cooldowns) { 0 }
     cattr_accessor :logger do
-      STDOUT.sync = true unless logdev = Settings.log.try(:file)
-      Logger.new(logdev || STDOUT, 10, 10240000).tap do |l|
-        l.level = Logger.const_get(Settings.log.level.upcase)
-        l.formatter = proc do |severity, datetime, progname, msg|
+      logdev = Settings.log.try(:file)
+      STDOUT.sync = true unless logdev.present?
+      Logger.new(logdev || STDOUT, 10, 10_240_000).tap do |log|
+        log.level = Logger.const_get(Settings.log.level.upcase)
+        log.formatter = proc do |severity, datetime, msg|
```

Comment:
This is broken, the proc is missing the `progname` argument.

Reply:
fixed.